webui/react/src/components/PageHeaderFoldable.tsx

```diff
@@ -77,18 +78,22 @@ const PageHeaderFoldable: React.FC<Props> = ({ foldableContent, leftContent, opt
             />
           )}
           <div className={css.optionsButtons}>
-            {options?.slice(0, 3).map((option) => (
-              <div className={css.optionsMainButton} key={option.key}>
-                <Button
-                  disabled={option.disabled || !option.onClick}
-                  icon={option?.icon}
-                  key={option.key}
-                  loading={option.isLoading}
-                  onClick={option.onClick}>
-                  {renderOptionLabel(option)}
-                </Button>
-              </div>
-            ))}
+            {options?.slice(0, 3).map((option) =>
+              option?.content ? (
+                option.content
```

Comment:
This seems a bit hacky in my opinion, do you think there might be a way around it? Such as `options?: (ButtonOption | DropdownOption)[]`?

Reply:
@gt2345 I could do that, I would then have to update several codepaths to handle either type which also felt a but hacky. 

To me it seems a little weird to force us to only show a "Button" or "Dropdown" here. It seems possible that one day we might want to show an icon, or freeform text, or some other component here. I can do this but this seemed like a more future-proof approach.

Reply:
@gt2345 I made several updates to this. We can now show any react node, and show multiple actions per react node, the latter was needed to show both of the actions in the button dropdown on the overflow list.